### PR TITLE
modified the colors, looks and added logic to the game

### PR DESCRIPTION
**Pull request by nknithi**
Please fill out the survey before submitting the pull request. Thanks!

🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

How many hours did you spend on this assignment? 10

Please fill in one error and/or error message you received while working on this assignment. Variable scope error.

What part of the assignment did you spend the most time on? Changing the players turn.

Comfort Level (1-5):3

Completeness Level (1-5):3

What did you think of this deliverable? Its fun and the project covers most of the topics learnt.

Is there anything in this code that you feel pleased about?

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -18,39 +18,51 @@
...
 
   <body>
     <h1 id="header">Basics: Beat That! 🚀</h1>
     <div id="container">
+      <p>Hello! Welcome to Beat That! Click submit to start the game.</p>
+      <p>
+        Create a two-digit number by selecting the order of your dice rolls.
+      </p>
+      <p>The player with the highest total score leads! Good luck!</p>
+
```
Comment (lee-kimmixq) [inline index.html:65]: Nice work amending HTML to give user instructions!

File: script.js [hunk 2]
```diff
@@ -1,4 +1,250 @@
...
+var currentPlayerTwoScore = 0;
+var winningPlayer = "";
+
+//function to roll a dice
+var rollADice = function () {
+  var randomNumber = Math.floor(Math.random() * 6) + 1;
+  return randomNumber;
+};
+//function to format the string
+var formatString = function (
```
Comment (lee-kimmixq) [inline script.js:21]: Very useful helper function to reduce cluttering of strings in the game logic! 👍

File: script.js [hunk 3]
```diff
@@ -1,4 +1,250 @@
...
+  message,
+  input,
+  combinedNumber,
+  totalPlayerOneScore,
+  totalPlayerTwoScore
```
Comment (lee-kimmixq) [inline script.js:26]: `combinedNumber`, `totalPlayerOneScore` and `totalPlayerTwoScore` are already global variables, so it is not necessary to include them in the arguments of the helper function. This will help to reduce code when you call the function, and also reduce the possibility of missing out 1 argument (which will affect the logiv of the code) as well!

File: script.js [hunk 4]
```diff
@@ -1,4 +1,250 @@
...
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  if (gameMode == "rollDiceMode") {
+    rollDiceMode(input);
+  } else if (gameMode == "resultMode") {
+    resultMode(input);
+  }
+
   return myOutputValue;
 };
```
Comment (lee-kimmixq) [inline script.js:250]: Very concise main function! 👍

File: script.js [hunk 5]
```diff
@@ -1,4 +1,250 @@
...
+};
+//function for first mode logic
+var rollDiceMode = function (input) {
+  if (input == "") {
+    randomNumberOne = rollADice();
+    randomNumberTwo = rollADice();
+
+    if (randomNumberOne == randomNumberTwo) {
+      equals = true;
+      combinedNumber = "" + randomNumberOne + randomNumberTwo;
```
Comment (lee-kimmixq) [inline script.js:112]: Interesting way of concatenating the 2 numbers!

File: script.js [hunk 6]
```diff
@@ -1,4 +1,250 @@
...
+        totalPlayerOneScore = totalPlayerOneScore + currentPlayerOneScore;
+      } else if (playerNumber == 2) {
+        currentPlayerTwoScore = Number(combinedNumber);
+        // playerTwoScores.push(currentPlayerTwoScore);
+        totalPlayerTwoScore = totalPlayerTwoScore + currentPlayerTwoScore;
+        if (currentPlayerOneScore > currentPlayerTwoScore) {
+          winningPlayer = "Player 1";
+        } else if (currentPlayerTwoScore > currentPlayerOneScore) {
+          winningPlayer = "Player 2";
+        }
```
Comment (lee-kimmixq) [inline script.js:125]: What will happen if both players draw?
You might want to include another condition for when both scores are the equal!

File: script.js [hunk 7]
```diff
@@ -1,4 +1,250 @@
...
+    } else if (playerNumber == 2) {
+      currentPlayerTwoScore = Number(combinedNumber);
+      // playerTwoScores.push(currentPlayerTwoScore);
+      totalPlayerTwoScore = totalPlayerTwoScore + currentPlayerTwoScore;
+      if (currentPlayerOneScore > currentPlayerTwoScore) {
+        winningPlayer = "Player 1";
+      } else if (currentPlayerTwoScore > currentPlayerOneScore) {
+        winningPlayer = "Player 2";
+      }
+    }
```
Comment (lee-kimmixq) [inline script.js:211]: This code is repeated from above, you might want to refactor it into a helper function!

File: script.js [hunk 8]
```diff
@@ -1,4 +1,250 @@
...
+
+      gameMode = "resultMode";
+    }
+  } else {
+    myOutputValue = formatString(
+      "click submit",
+      input,
+      totalPlayerOneScore,
+      totalPlayerTwoScore
+    );
```
Comment (lee-kimmixq) [inline script.js:187]: Although the function `formatString()` takes in 5 arguments you don't need to fill in all of them if they are not going to be used in the function. In this case, if the first argument (`message`) is `"click submit"`, `formatString()` does not use any of the other arguments in its output. So, you can shorten this to `myOutputValue = formatString("click submit")` and it will still work the same!

**Review by lee-kimmixq — COMMENTED** [review]
Great work Nithya!
Your code logic shows that you have understood if conditions, global variables and game modes, and good use of helper functions to make your main function more concise! I really liked your `formatOutput()` function and also how you had an if condition to check if both dice rolls were the same to automatically concatenate the 2 numbers!

Some suggestions to improve and make your code easier to read are:
1. Further refactor your code (especially code that is repeated)
2. Use `myOutputValue += ….` instead of `myOutputValue = myOutputValue + …`
3. Reduce arguments for helper functions (see code comments)
4. Use [template literals](https://basics.rocketacademy.co/4-getting-started-with-code/4.2-our-first-program#output-formatting) to format your strings! It reduces the possibility of syntax error due to missing + signs or quotation marks (as well as makes your code shorter and more readable)

**Comment by nknithi** [comment]
Thanks a lot for your detailed review Kimmi. I have taken note of all your suggestions and will try to follow it for my future projects. Thanks again. You are one of the best....😊👍
